common-content/en/module/decomposition/backend-limitations/index.md

```diff
@@ -0,0 +1,57 @@
++++
+title = "Limitations of backends"
+headless = true
+time = 20
+facilitation = false
+emoji= "📖"
+objectives = [
+    "Explain why a backend on its own doesn't provide reliable data persistence.",
+    "Explain why we may prefer doing work in a frontend rather than backend to avoid latency.",
+]
++++
+
+We've already explored limitations of frontends.
+
+We know a backend is just a program that runs for a long time.
+
+### Lifetime
+
+A major limitation of backends is that "a long time" probably isn't forever.
+
+Sometimes we change the code of the backend. Or need restart the computer it's running on for an upgrade. Or its computer loses power and we need to start it again.
+
+When this happens, the program starts again.
+
+Think back to our quote server that allows users to POST new quotes.
+
+If we had to stop the server and start it again, we would lose all of the quotes users had saved.
+
+They're just stored in a variable, and a variable only lasts while the program it's in is running.
+
+### Location
+
+Another major limitation of a backend is where the code runs.
+
+A backend's code runs on whatever server it's running on.
+
+In contrast, a web frontend's code runs in the user's web browser.
+
+#### Latency
+
+One problem here is latency. Depending on where the backend and the user are physically located, it may take anywhere between 1ms and 500ms for a request to go between them.
```

Comment:
lead with a concrete example. We have already talked about latency -- you can bring that definition in here and reuse it if you want. But make a clear bridge here to prior learning.